src/libraries/System.Text.Json/src/System/Text/Json/Serialization/WriteStackFrame.cs

```diff
@@ -28,6 +28,8 @@ internal struct WriteStackFrame
         public bool StartObjectWritten;
         public bool MoveToNextProperty;
 
+        public bool WriteWrappingBraceOnEndCollection;
```

Comment:
```suggestion
        public bool WriteWrappingBraceOnEndPreservedArray;
```

since this only applies to preserved JSON arrays